Kernel/Graphics/VirtIOGPU/GPU3DDevice.h

```diff
@@ -130,13 +143,15 @@ class VirtIOGPU3DDevice : public CharacterDevice {
     virtual ErrorOr<void> ioctl(OpenFileDescription&, unsigned request, Userspace<void*> arg) override;
     virtual void detach(OpenFileDescription&) override;
 
+    using ContextsList = IntrusiveListRelaxedConst<&PerContextState::m_list_node>;
+
 private:
-    ErrorOr<LockRefPtr<PerContextState>> get_context_for_description(OpenFileDescription&);
+    ErrorOr<NonnullRefPtr<PerContextState>> get_context_for_description_from_unlocked_list(ContextsList& list, OpenFileDescription& description);
```

Comment:
Maybe instead of having an API that requires the list to be unlocked (which is also not verified at the moment), why not have a method that looks up the description, does the unlocking itself and then invokes a lambda with the description as an argument?

Reply:
We could do that, or just instead put the entire function inside the ioctl method as lambda function, which I prefer more :)